Docs: api keys (#64803)
grafana/grafana: docs/sources/auth/planning/index.md

### 🚧 Personal access tokens

### API keys

> **Note:** If you use Grafana v8.5 or newer, you should use service accounts instead of API keys. API keys will be deprecated in the near future. For more information, refer to [Grafana service accounts]({{< relref "../service-accounts/" >}}).

> You can use Grafana API keys to interact with data sources via HTTP APIs. API keys can have a well-defined and limited scope to resources with the help of [Roles].

Reviewer: API keys aren't limited to data sources - they can be used to interact with any Grafana resource.

### 🚧 API keys

> Grafana API keys can be used to interact with data sources via HTTP APIs. API keys can have a well-defined and limited scope to resources with the help of [Roles].

Reviewer 2: ```suggestion
You can use Grafana API keys to interact with data sources via HTTP APIs. API keys can have a well-defined and limited scope to resources with the help of [Roles].
```

Author: Changed!

> Grafana API keys can be used to interact with data sources via HTTP APIs. API keys can have a well-defined and limited scope to resources with the help of [Roles].

Reviewer 2: Should we point users to the service account docs as the way to do this? Eric has made some nice updates to those docs recently (not sure if merged in yet).

> Grafana API keys can be used to interact with data sources via HTTP APIs. API keys can have a well-defined and limited scope to resources with the help of [Roles].

Reviewer 2: I'm not sure what "to resources with the help of roles" means.

Author: I meant a limited scope or limited access to resources (editing dashboards, viewing sensitive information) with a well-defined role/permission definition.

If this is not clear at first glance, I should reword it.

Reviewer: You can only use the basic roles (Viewer, Editor, Admin) for API keys. Service accounts are more flexible and Enterprise users can assign specific RBAC roles to them.

> > **Note:** If you use Grafana v8.5 or newer, use service accounts instead of API keys. For more information, refer to [Grafana service accounts]({{< relref "../service-accounts/" >}}).

Reviewer 2: oh nvm, you do point to the service account docs. :)

Reviewer: I actually think that we should push this info higher (maybe even in the first sentence?) and be very explicit that API keys are deprecated in favour of service accounts.

Author: I moved and rephrased the warning to the top level.
